sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1221,4 +1221,48 @@ export class ApplicationService extends RecordDataModelService<Application> {
       })
       .getMany();
   }
+  /**
+   * Gets active application.
+   * @param applicationId application id.
+   * @param locationId location id.
+   * @returns active application details.
+   */
+
+  async getActiveApplication(
+    applicationId: number,
+    locationId: number,
+  ): Promise<Application> {
+    return this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationStatus",
+        "application.applicationNumber",
+        "offering.offeringIntensity",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "location.name",
+        "offering.name",
+        "pirProgram.name",
+        "pirProgram.description",
+        "student",
+        "user.firstName",
+        "user.lastName",
+      ])
+      .leftJoin("application.offering", "offering")
+      .leftJoin("application.pirProgram", "pirProgram")
+      .leftJoin("application.location", "location")
```

Comment:
Applications completed must have a location. I would recommend changing it to an inner join.